### DISCUSSION TITLE
请问如何加载并保留forked仓库中的image.config配置？

### DISCUSSION THREAD

**Discussion by wanmyj** [chenmozhijin/OpenWrt-K#47] [Q&A] [answered]
清空配置环境，使用OpenWrt‐k配置构建工具去生成config文件夹的配置，repo填的是我自己forked的仓库，关于“你是否要修改openwrt的TARGET配置“无论选择是 或者 否，生成的image.config相比于repo的image.config都会发生改变。

我看到openwrt的配置菜单里的配置项是如下图:
![image](https://github.com/chenmozhijin/OpenWrt-K/assets/41611679/8a21b3dc-44f7-40e0-a615-033a833339b9)

我个人仓库的image.config如下
<details>
  <summary>image.config</summary>

```config
#
# Target Images
#
#
# Root filesystem archives
#
#
# Root filesystem images
#
CONFIG_TARGET_ROOTFS_SQUASHFS=n
CONFIG_TARGET_ROOTFS_EXT4FS=y
CONFIG_TARGET_ROOTFS_TARGZ=n
CONFIG_GRUB_IMAGES=n
CONFIG_GRUB_TIMEOUT="0"
CONFIG_VHDX_IMAGES=y
# CONFIG_ISO_IMAGES=y
# CONFIG_VMDK_IMAGES=y

#
# Image Options
#
CONFIG_TARGET_KERNEL_PARTSIZE=256
CONFIG_TARGET_ROOTFS_PARTSIZE=4096
# end of Target Images
```
</details>

预期：生成的config文件夹里的image.config与个人仓库相同（我并没有打算去改变image.config的内容）
实际：
```
sh@sh-vm:~/openwrt_config_gen/config/x86_64$ cat image.config
#
# Target Images
#
#
# Root filesystem archives
#
CONFIG_TARGET_ROOTFS_TARGZ=n
#
# Root filesystem images
#
CONFIG_TARGET_ROOTFS_SQUASHFS=n
CONFIG_GRUB_IMAGES=n
CONFIG_GRUB_TIMEOUT="0"
CONFIG_VHDX_IMAGES=y
#
# Image Options
#
CONFIG_TARGET_KERNEL_PARTSIZE=256
CONFIG_TARGET_ROOTFS_PARTSIZE=4096
# end of Target Images
```

**Comment by wanmyj** [comment]
研究了下，这个好像是没办法的，使用构建工具会覆盖自定义的配置，只能在git add的时候把image.config去掉。

**Comment by chenmozhijin** [comment, marked as answer]
这应该不是一个bug,OpenWrt-K配置构建工具生成的配置由配置差异文件制作而来，它只包含更改（见openwrt官网使用配置文件差异文件进行配置）。

所以被删掉的那几行应该并不会改变你的编译结果，它们只是与openwrt官方的配置相同，所以直接被忽略掉了。

**Reply by wanmyj** [reply]
> 它们只是与openwrt官方的配置相同，所以直接被忽略掉了。

很关键的信息，多谢。👌
